[6.1] Github Codespaces Patch

mahmoudmagdy1-1 opened the pull request:
Pull Request for Issue # .

### Summary of Changes
This PR is a quick patch for github codespaces feature, #45719
1. it provides https on port 443, which is needed for some joomla features and it caused `Mfa.cy.js` tests to fail 
<img width="1313" height="695" alt="image" src="https://github.com/user-attachments/assets/28464523-8f00-4d59-9d4c-5a4db8cc051a" />
2. readds `rm configuration.php` to rebuild container without any errors
3. Adds error logging for php


### Testing Instructions
see https://github.com/joomla/Manual/pull/496


### Actual result BEFORE applying this Pull Request
`Mfa.cy.js` tests fail, error when rebuilding the container (it says configuation.php already exists), no error logs for php


### Expected result AFTER applying this Pull Request
`Mfa.cy.js` tests pass and https support is added, no errors when rebuilding the container, error logs available for php



### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [ ] No documentation changes for docs.joomla.org needed

- [x] Pull Request link for manual.joomla.org: https://github.com/joomla/Manual/pull/505
- [ ] No documentation changes for manual.joomla.org needed

mahmoudmagdy1-1 committed:
Github Codespaces Patch

laoneo commented:
Would this mean that it works only on https? Does it also need an update of the manual?

mahmoudmagdy1-1 commented:
> Would this mean that it works only on https?
>
It was working only on http (port 80), with this pr it works on both http (port 80) and https (port 443)

> Does it also need an update of the manual?
>
I guess so, just a few lines to clarify.
I'll make a pr for that later today

Commit:
Change php logs / changed Router cy tests

mahmoudmagdy1-1 commented:
Just added a couple of lines in the docs for this, https://github.com/joomla/Manual/pull/505
> there are still 2 failure about Router.cy.js
I changed them like your comments
<img width="673" height="249" alt="image" src="https://github.com/user-attachments/assets/c3172f83-74ac-406f-8110-60df5a01194f" />
now we got all Cypress tests passing successfully
> what i've noticed is that the configuration.php is not writable so you cannot change some settings in Global configuration
After some debugging, I found out it was caused by `SiteUpDown.cy.js` and affected by `cli/joomla.php site` commands

mahmoudmagdy1-1 commented:
> Do we need port 3306 for MySQL to the outside? 

Yes, otherwise users can't execute commands from the codespace terminal like in the image below, and in any way I don't think there is a down side from having them exposed, as a github codespace is only accessible to the github account who made the codespace and also because it's a testing environment

<img width="756" height="76" alt="image" src="https://github.com/user-attachments/assets/77b97a63-61c5-4c05-9337-e360589def1d" />

Thank you for the suggestion

mahmoudmagdy1-1 committed:
Reverse Router.cy.js & Fix unwritable config after Cypress tests

bembelimen committed:
Merge branch '6.0-dev' into gh-codespaces-patch

bembelimen committed:
Merge branch '6.1-dev' into gh-codespaces-patch
